drivers/spi/spi_nrfx_spim.c

```diff
@@ -310,25 +311,34 @@ static void transfer_next_chunk(const struct device *dev)
 		nrfx_spim_xfer_desc_t xfer;
 		nrfx_err_t result;
 		const uint8_t *tx_buf = ctx->tx_buf;
-#if (CONFIG_SPI_NRFX_RAM_BUFFER_SIZE > 0)
-		if (spi_context_tx_buf_on(ctx) && !nrfx_is_in_ram(tx_buf)) {
-			if (chunk_len > CONFIG_SPI_NRFX_RAM_BUFFER_SIZE) {
-				chunk_len = CONFIG_SPI_NRFX_RAM_BUFFER_SIZE;
-			}
+		uint8_t *rx_buf = ctx->rx_buf;
 
-			memcpy(dev_data->buffer, tx_buf, chunk_len);
-			tx_buf = dev_data->buffer;
-		}
-#endif
 		if (chunk_len > dev_config->max_chunk_len) {
 			chunk_len = dev_config->max_chunk_len;
 		}
 
+#ifdef SPI_BUFFER_IN_RAM
+		if (chunk_len > CONFIG_SPI_NRFX_RAM_BUFFER_SIZE) {
+			chunk_len = CONFIG_SPI_NRFX_RAM_BUFFER_SIZE;
+		}
```

Comment:
This should not be applied always, but only when a given transfer actually uses the internal buffer.